src/libsystemd-network/sd-dhcp-client.c

```diff
@@ -576,13 +576,16 @@ int sd_dhcp_client_set_user_class(
                 const char* const* user_class) {
 
         _cleanup_strv_free_ char **s = NULL;
-        char **p;
+        const char * const *p;
 
-        STRV_FOREACH(p, (char **) user_class)
-                if (strlen(*p) > 255)
-                        return -ENAMETOOLONG;
+        STRV_FOREACH(p, user_class) {
+                size_t n = strlen(*p);
 
-        s = strv_copy((char **) user_class);
+                if (n > 255 || n == 0)
```

Comment:
Does strv allows 0 length strings ?

Reply:
Yes. So, we need to drop zero length entries explicitly.